src/main/java/bio/terra/app/controller/DatasetsApiController.java

```diff
@@ -150,6 +150,13 @@ public ResponseEntity<DatasetModel> retrieveDataset(
     AuthenticatedUserRequest userRequest = getAuthenticatedInfo();
     iamService.verifyAuthorization(
         userRequest, IamResourceType.DATASET, id.toString(), IamAction.READ_DATASET);
+    if (include.contains(DatasetRequestAccessIncludeModel.SNAPSHOT_BUILDER_CONFIG)) {
```

Comment:
I thought this might belong here, since its IAM related? But it also contains some logic, so I could see pushing it down - we just might need to plumb the iamService through to the relevant spot.

Reply:
Yeah, we unfortunately haven't been super consistent here but this lines up (we're doing another check right above, so this makes sense)

Reply:
Hey!  I think this is the right place for auth checks, but you might find it helpful to see this existing case where we conditionally verify multiple IAM actions depending on the user's request:

https://github.com/DataBiosphere/jade-data-repo/blob/develop/src/main/java/bio/terra/app/controller/SnapshotsApiController.java#L153-L154

I prefer that this makes only one call to Sam to check all the needed actions (like what @pshapiro4broad said), and we can unit test the actions required in a targeted way.